src/Compilers/Test/Core/Compilation/TestOperationVisitor.cs

```diff
@@ -578,7 +578,8 @@ private void VisitMemberReference(IMemberReferenceOperation operation, IEnumerab
                 if (operation.Member.IsStatic &&
                     operation.Instance is IInstanceReferenceOperation)
                 {
-                    Assert.False(operation.Instance.IsImplicit, $"Implicit {nameof(IInstanceReferenceOperation)} on {operation.Syntax}");
+                    Assert.True(!operation.Instance.IsImplicit || operation.Instance.HasErrors(operation.SemanticModel.Compilation),
```

Comment:
Aleksey commented:

> It feels like this breaks one of the invariants that we have. Why the original condition was fine for a regular member access and a problem for patterns?
> It feels to me that we should fix the product instead and maintain the invariant.

Reply:
> Why the original condition was fine for a regular member access and a problem for patterns?

Because the implementation included a receiver unconditionally. I'll adjust the implementation to not do so and restore this.

---
In reply to: [551545224](https://github.com/dotnet/roslyn/pull/50111#discussion_r551545224) [](ancestors = 551545224)